### PR TITLE
29/06/2024 trueup api

### PR DESCRIPTION
**Pull request by daniel-sussman**

### CODE REVIEW

File: app/models/concerns/ats/trueup/company_details.rb
Comment (Ches-ctrl) [inline app/models/concerns/ats/trueup/company_details.rb:1]: Think in future we want to handle job_boards separately (i.e. with different nested modules) but for now think this does the job

File: app/services/faraday_helpers.rb
Comment (Ches-ctrl) [inline app/services/faraday_helpers.rb:1]: Can you help me understand the restructure here? Will this break any existing usage of Faraday?

File: spec/fixtures/cassettes/true_up.yml
Comment (Ches-ctrl) [inline spec/fixtures/cassettes/true_up.yml:1]: What is this cassette saving at the moment? Looks short

File: app/services/importer/api/devit_jobs.rb
Comment (Ches-ctrl) [inline app/services/importer/api/devit_jobs.rb:1]: Can you help me understand this restructure a little more?

File: storage/19/5g/195gx2gcdc3cr3aomgdpkid7zw3n
Comment (Ches-ctrl) [inline storage/19/5g/195gx2gcdc3cr3aomgdpkid7zw3n:1]: We need to turn off saving these temporary files going forward - think it's a JB setting

File: storage/csv/ats_systems.csv
Comment (Ches-ctrl) [inline storage/csv/ats_systems.csv:1]: TODO: Add separate job_board list (later)

File: storage/csv/unparseable_urls.csv
Comment (Ches-ctrl) [inline storage/csv/unparseable_urls.csv:1]: This files becoming quite large - just one to note for later (need to make sure we're not uploading it to heroku to sit there taking up slug space)

File: storage/19/5g/195gx2gcdc3cr3aomgdpkid7zw3n
Comment (Ches-ctrl) [reply storage/19/5g/195gx2gcdc3cr3aomgdpkid7zw3n:1]: Can delete these manually now or later in the meantime

**Review by Ches-ctrl — COMMENTED** [review]
Adding comments for discussion

File: app/models/concerns/ats/trueup/company_details.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+        data['company_id']
+      end
+
+      def company_details(ats_identifier)
+        data = Importer::Scraper::TrueUpCompanyDetails.call(ats_identifier)
+        {
+          name: data['company'],
+          url_website: data['company_url_clean'], # TODO: make this the long form
+          description: Flipper.enabled?(:company_description) ? data['business_description_long'] : 'Not added yet',
+          industry: data['company_classification'],
```
Comment (patanj101) [inline app/models/concerns/ats/trueup/company_details.rb:15]: maybe put the logic in a method ... fetch_description or so

File: app/models/concerns/ats/trueup/job_details.rb [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+      def build_description(data)
+        title = data['title']
+        company_name = data['company_short']
+        business_description = data['business_description_short']
+        location = data['location']
+        apply_url = data['url']
+        "<p>#{company_name}, a #{business_description&.downcase} company, is hiring a #{title} in #{location}.</p><p>To apply, visit <a href=\"#{apply_url}\">#{apply_url}</a>.</p>"
+      end
+
+      def build_salary(data)
```
Comment (patanj101) [inline app/models/concerns/ats/trueup/job_details.rb:40]: Small refactoring ... Just a suggestion. I did not test it.

```
def build_salary(data)
  min, max = extract_salary_range(data)
  return unless min && max && min.is_a?(Numeric) && max.is_a?(Numeric)

  salary_low = number_with_delimiter(min)
  salary_high = number_with_delimiter(max)

  salary_range = min == max ? "#{salary_low}" : "#{salary_low} - £#{salary_high}"

  "£#{salary_range} GBP"
end

private

def extract_salary_range(data)
  min = data['salary_range_min']
  max = data['salary_range_max']

  [min, max]
end
```

**Review by patanj101 — APPROVED** [review]

**Pull request merged**